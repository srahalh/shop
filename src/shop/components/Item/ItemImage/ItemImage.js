import React from 'react';
import PropTypes from 'prop-types';

import './ItemImage.scss';

const ItemImage = ({imageURL}) => {
  return (
    <img
      className="ItemImage__container"
      alt="Some images here"
      src={imageURL}
    />
  )
}

ItemImage.propTypes = {
  imageURL: PropTypes.string
}

export default ItemImage