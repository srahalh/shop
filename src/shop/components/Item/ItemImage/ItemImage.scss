.ItemImage__container {
  height: 180px;
  width: 100%;
  display: block;
  object-fit: contain;
}