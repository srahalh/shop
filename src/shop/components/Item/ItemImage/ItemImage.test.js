import React from "react";
import { render } from "@testing-library/react";
import ItemImage from "./ItemImage";
const products = require("../../../../../mock/products.json");

describe("ItemImage component", () => {
  test("Should render with imgURL", () => {
    const imageURL = products[0].imgUrl;
    const { asFragment  } = render(<ItemImage imageURL={imageURL} />);
    expect(asFragment ()).toMatchSnapshot();
  });
});