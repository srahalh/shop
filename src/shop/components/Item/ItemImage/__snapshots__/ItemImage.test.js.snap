// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemImage component Should render with imgURL 1`] = `
<DocumentFragment>
  <img
    alt="Some images here"
    class="ItemImage__container"
    loading="lazy"
    src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
  />
</DocumentFragment>
`;
